lib/person.rb

```diff
@@ -0,0 +1,26 @@
+class Person
+  attr_reader :name,
+              :interests,
+              :supplies
+  def initialize(info)
+    @name = info[:name]
+    @interests = info[:interests]
+    @supplies = {}
+  end
+
+  def add_supply(supply, amount)
+    if @supplies.keys.include?(supply) == false
+      @supplies[supply] = amount
+    else
+      @supplies[supply] += amount
+    end
+  end
+
+  def can_build?(craft)
```

Comment:
You would want to iterate through `craft.supplies_required` to check if a particular supply(key) has more than or equal to the  quantity needed for that craft. So a possible solution could be:
```
def can_build?(craft)
    supply_check = craft.supplies_required.map do |supply, quantity|
      @supplies[supply.to_s] >= quantity
    end
    supply_check.all?
  end
end
```
 If you put a pry in above `supply_check.all?`, the first time through it would return [false, false, false, false]. The second time through, you add `fabric, 7` and `thread, 1`, so it would return [true, false, true, false]. The third time through, you add `scissors, 1` and `sewing_needles, 1`, so it would return [true, true, true, true].